### PR TITLE
DEPR: deprecate unit parameters ’T’, 't', 'L', and 'l'

### LINKED ISSUE #52536: BUG: Either incorrect unit validation for 'T' in to_timedelta() or incorrect documentation
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
td = pd.to_timedelta(1, unit='t')  # success
td = pd.to_timedelta(1, unit='T')  # runs ok but fails type-checking in IDE
```

### Issue Description

The specification of the unit parameter for to_timedelta() in https://github.com/pandas-dev/pandas/tree/v2.0.0/pandas/core/tools/timedeltas.py is as follows:

`unit: UnitChoices | None = None`

The [documentation](https://pandas.pydata.org/docs/reference/api/pandas.to_timedelta.html) for to_timedelta() (and the comment in the code) allows for (among others):

> ‘m’ / ‘minute’ / ‘min’ / ‘minutes’ / ‘T’

The allowed values in UnitChoices (https://github.com/pandas-dev/pandas/tree/main/pandas/_libs/tslibs/timedeltas.pyi) include 't', but not 'T'.

### Expected Behavior

It seems to me that it is usual to allow upper case 'T' as a specifier for minute, but others may know better.

Either 'T' should be allowed in UnitChoices, or the to_timedelta() documentation should be updated to specify lower case 't'

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 478d340667831908b5b4bf09a2787a11a14560c9
python           : 3.9.6.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 22.4.0
Version          : Darwin Kernel Version 22.4.0: Mon Mar  6 21:00:17 PST 2023; root:xnu-8796.101.5~3/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : en_GB.UTF-8
pandas           : 2.0.0
numpy            : 1.24.2
pytz             : 2023.3
dateutil         : 2.8.2
setuptools       : 65.5.1
pip              : 22.3.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.7.1
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.10.1
snappy           : None
sqlalchemy       : 1.4.47
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
zstandard        : None
tzdata           : 2023.3
qtpy             : None
pyqt5            : None

</details>

### PR DESCRIPTION
**Pull request by natmokval**
- [x] closes #52536
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Added upper case ’T’ for minutes in `UnitChoices` list. Now mypy doesn’t raise an error.

Update:
Deprecated unit parameters ’T’, 't', 'L', 'l', and fixed test.

### CODE REVIEW

**Comment by natmokval** [comment]
@jbrockmendel, could you please take a look at this pr?

File: pandas/core/tools/timedeltas.py [hunk 1]
```diff
@@ -181,6 +181,9 @@ def to_timedelta(
             "represent unambiguous timedelta values durations."
         )
 
+    if unit in {"t", "T", "l", "L"}:
+        raise ValueError("Units 't', 'T', 'l' and 'L' are no longer supported.")
```
[the remark this answers is not in the document]
Comment (natmokval) [reply pandas/core/tools/timedeltas.py:185]: Thanks for the comment. I’ll replace raising ValueError with FutureWarning.

**Comment by jbrockmendel** [comment]
along with these i think we need to track down other places where we use/allow these codes.  _attrname_to_abbrevs/_abbrev_to_attrnames and Tick._prefix come to mind.  Doesn't need to be this PR, but hopefully a follow-on

**Review by MarcoGorelli — APPROVED** [review]
Nice one!

Test failures are probably unrelated, if we fetch/merge upstream/main after they're fixed then hopefully it'll all be good

**Comment by natmokval** [comment]
Thank you @MarcoGorelli for reviewing this pr.

**Comment by MarcoGorelli** [comment]
merging then, thanks!

**Comment by cewing123** [comment]
Why do this? This is a pain in the ass for people

**Comment by MarcoGorelli** [comment]
> Why do this? This is a pain in the ass for people

looks like it comes from [here](https://github.com/pandas-dev/pandas/issues/52536#issuecomment-1581655087)

**Pull request merged**